File: oss-file-service/src/main/java/com/oss/service/RoleService.java
package com.oss.service;

import com.github.pagehelper.Page;
import com.oss.model.Role;
import com.oss.pojo.dto.RoleListDto;
import com.oss.pojo.vo.RoleVo;
import com.oss.tool.BaseService;
import com.oss.pojo.dto.RoleAllotDto;
import com.oss.pojo.dto.ZoneAllotDto;
import com.oss.tool.ResponseResult;

import java.util.List;

/**
 * @InterfaceName：RoleService
 * @Description: 角色服务
 * @Author：13738700108
 * @Data 2020/11/3 0:05
 * @Version: v1.0
 **/
public interface RoleService extends BaseService {

    /**
     * 查询角色列表
     * @return
     */
    ResponseResult<Page<RoleVo>>  getRoleList(RoleListDto roleListDto);

    /**
     * 分配权限
     * @param roleAllotDto
     * @return
     */
    ResponseResult allotRole(RoleAllotDto roleAllotDto);

    /**
     * 分配分区
     * @param zoneAllotDto
     * @return
     */
    ResponseResult allotZone(ZoneAllotDto zoneAllotDto);


    /**
     * 初始化项目
     * @return
     */
    ResponseResult start();
}

File: oss-file-web/src/service/RoleService.js
import http from '@/util/http'

/** 查询我的角色所有密匙*/
const getMyRolePwd = (param = {}) =>{return http.post({url: '/role/getMyRolePwd', data: param})}

/** 输入角色密钥*/
const getRoleKey = (param = {}) =>{return http.post({url: '/role/getRoleKey', data: param})}

/** 分配分区*/
const allotZone = (param = {}) =>{return http.post({url: '/role/allotZone', data: param})}

/** 分配角色*/
const allotRole = (param = {}) =>{return http.post({url: '/role/allotRole', data: param})}

/** 查询角色列表*/
const getRoleList = (param = {}) =>{return http.post({url: '/role/getRoleList', data: param})}


export default {
  getMyRolePwd,                                          /** 查询我的角色所有密匙*/
  getRoleKey,                                            /** 输入角色密钥*/
  allotZone,                                            /** 分配分区*/
  allotRole,                                            /** 分配角色*/
  getRoleList,                                          /** 查询角色列表*/
}
